proto/orchestrator.proto

```diff
@@ -79,6 +79,19 @@ message SubmitProofRequest {
 
   // The task's ID.
   string task_id = 6;
+
+  // A Ed25519 public key (~32 bytes) generated by
+  // the node to uniquely identify this request,
+  // used for DoS protection.
+  bytes ed25519_public_key = 7;
+
+  // A signature of the proof, generated by signing
+  // task_id + hash(proof) with the Ed25519 private key
+  // corresponding to the public key.
+  bytes signature = 8;
+
+  // Signature Version
+  int32 signature_version = 9;
```

Comment:
@collinjackson The ed25519_public_key and signature fields are from your PR. I've added the signature_version field to help with backwards compatibility and domain separation.

Reply:
I think we should cross this bridge when we actually want to support multiple versions at the same time (not sure we will any time soon)